### DISCUSSION TITLE
Troubleshooting Help, 8080 Won't Start

### DISCUSSION THREAD

**Discussion by Gmanxxx** [thehighnibble/thehighnibble.github.io#127] [IMSAI8080esp replica]
My 8080 won't start after final assembly.  I tested it as per THN#19 by inserting the power switch and was able to get CPM to load and run.  After soldering all of the switches and assembling the complete unit, when I turn on the power switch, none of the lights come on.

At first I thought I damaged the switch by soldering, but using an ohmmeter, it seems to be functioning OK.
- The center pad of the switch stays at 5V regardless of the switch position.  I am not sure if the voltage is supposed to go to 
    ground.  Is that normal?
-Somehow the uSD socket got dislodged and I soldered it back down.  When I check the log, there is an error message about
    the uSD card (attached).  I checked and re-soldered all of the connections and still get the error message.

Any suggestions?
[teraterm.log](https://github.com/thehighnibble/thehighnibble.github.io/files/13596393/teraterm.log)

**Comment by nbreeden2** [comment]
It sounds like the power switch is fine, getting the log on the serial terminal when you apply power indicates the switch is working. This would indicate the firmware in the ESP32 is working.

When you press RESET do all the LEDs light up?

Have you checked the uSD card on a PC? You should see a few folders at the root.
CROMEMCO
IMSAI
WWW
Each should have folders/files in them.

Please attach some close-up photos of the soldering for the ESP32 and the uSD connector.

**Comment by Gmanxxx** [comment]
Thank you.  I think the problem is the uSD card.  

I connected it to multiple PC using multiple adapters and none of them even registered the card.  I have a 32GB card available and hope there is an online set of files I can download.  A damaged SD card seems strange because the system was operational, I guess I did something that damaged the card itself during the final assembly.

To answer the other questions:
1. No, the LEDs don't light up on reset.
2. Pictures attached.

![image](https://github.com/thehighnibble/thehighnibble.github.io/assets/152238153/faec4f8c-c1f6-43ac-a1a3-01c367604afa)

![image](https://github.com/thehighnibble/thehighnibble.github.io/assets/152238153/742f4533-eaad-4d4f-8db6-71a27ddd8dcc)

**Comment by nbreeden2** [comment]
You'll find the version info and links to downloads here:
https://github.com/thehighnibble/firmware/releases

**Comment by Gmanxxx** [comment]
Thanks.  My unit is several years old (early 2020, yeah I am a procrastinator) so I assume I need to go to 1.10.0 first and not knowing what version is in my ESP, I need to update both it and the uSD to 1.10.0.  Using the desktop UI doesn't seem to be an option, so I need to do this the hard way with the esp.py tool?

**Comment by Gmanxxx** [comment]
Success.   I was able to figure out my ESP32 had version 1.5 downloaded, so I downloaded Ver 1.5 of the IMAGE and am back to running CP/M.  Interesting in the USB log, it identifies as Ver 1.37.  Not sure why.

**Comment by nbreeden2** [comment]
Awesome, glad it's working. :)

**Comment by Gmanxxx** [comment]
Was able to bootstrap from 1.5 to 1.10 to the current 1.12.  Everything seems A-OK.
